### PR TITLE
Augment citrine-python FileLinks with platform metadata

### PR DESCRIPTION
**Pull request by kroenlein**
# Citrine Python PR

## Description 
This PR augments the FileLink objects returned from the files endpoints to include the full metadata included in those responses.

https://citrine.atlassian.net/browse/PLA-10989

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in __version\__.py

### CODE REVIEW

**Comment by kroenlein** [comment]
e2e tests: https://jenkins.corp.citrine.io/job/nextgen-ITs/36059

File: src/citrine/resources/file_link.py [hunk 1]
```diff
@@ -639,14 +588,14 @@ def _complete_upload(self, dest_name: str, uploader: _Uploader):
                                                       json={'s3_version': uploader.s3_version})
 
         try:
+            print(complete_response['file_info'])
```
Comment (bfolie) [inline src/citrine/resources/file_link.py:591]: do you want to keep this print statement?

Comment (kroenlein) [reply src/citrine/resources/file_link.py:591]: No 🙇

File: src/citrine/resources/file_link.py [hunk 2]
```diff
@@ -217,83 +226,23 @@ def _get_path_from_file_link(self, file_link: FileLink,
...
+            file_id, version_id = self._get_ids_from_url(file_link.url)
+            if file_id is None:
+                raise ValueError("FileLink did not contain a Citrine platform file URL.")
         return self._get_path(uid=file_id, version=version_id, action=action)
 
     def build(self, data: dict) -> FileLink:
         """Build an instance of FileLink."""
+        # Use this chance to construct a URL from platform metadata
+        if 'url' not in data:
+            data['url'] = self._get_path(uid=data["id"], version=data["version"])
```
Comment (kroenlein) [inline src/citrine/resources/file_link.py:243]: A reasonable alternative approach here would be to have the collection builder insert a `project` and `dataset` field and then have the object builder use those to assemble a URL.  That probably matches responsibilities better, but would presumably have the FileLink access either `_get_path` or `_path_template`, which would technically be a private space violation.

**Pull request merged**